web/core/components/dropdowns/layout.tsx

```diff
@@ -0,0 +1,54 @@
+import { observer } from "mobx-react";
+import { Check } from "lucide-react";
+import { cn } from "@plane/editor";
+import { Dropdown } from "@plane/ui";
+// constants
+import { EIssueLayoutTypes, ISSUE_LAYOUT_MAP } from "@/constants/issue";
+
+type TLayoutDropDown = {
+  onChange: (value: EIssueLayoutTypes) => void;
+  value: EIssueLayoutTypes;
+};
+
+export const LayoutDropDown = observer((props: TLayoutDropDown) => {
+  const { onChange, value = EIssueLayoutTypes.LIST } = props;
+
+  const options = Object.values(ISSUE_LAYOUT_MAP).map((issueLayout) => ({
+    data: issueLayout.key,
+    value: issueLayout.key,
+  }));
+
+  return (
+    <Dropdown
+      onChange={onChange as (value: string) => void}
+      value={value?.toString()}
+      keyExtractor={(option: any) => option.value}
+      options={options}
+      buttonContainerClassName="bg-custom-background-100 border border-custom-border-200 hover:bg-custom-background-90 focus:text-custom-text-300 focus:bg-custom-background-90 px-2 py-1.5  rounded flex items-center gap-1.5 whitespace-nowrap transition-all justify-center relative"
+      buttonContent={(isOpen: boolean, buttonValue: string | string[] | undefined) => {
```

Comment:
Lets not declare these inline. Define them inside the component and wrap then in useCallback.